src/cpu/aarch64/injectors/jit_uni_eltwise_injector.cpp

```diff
@@ -2411,24 +2423,29 @@ void jit_uni_eltwise_injector_f32_t<isa>::register_table_entries() {
 }
 
 template <>
-size_t jit_uni_eltwise_injector_f32_t<asimd>::get_vec_len() {
+size_t jit_uni_eltwise_injector_t<asimd>::get_vec_len() {
     return 16;
 }
 
 template <>
-size_t jit_uni_eltwise_injector_f32_t<sve_128>::get_vec_len() {
+size_t jit_uni_eltwise_injector_t<sve_128>::get_vec_len() {
     return get_sve_length();
 }
 
 template <>
-void jit_uni_eltwise_injector_f32_t<asimd>::relu_zero_ns_compute_vector_fwd(
-        const TRegS &vmm_src) {
-    h->movi(vmm_aux0, 0);
-    h->fmaxnm(vmm_src, vmm_src, vmm_aux0);
+void jit_uni_eltwise_injector_t<asimd>::relu_zero_ns_compute_vector_fwd(
+        const TReg &vmm_src) {
+    if (d_type_ == data_type::f16) {
+        h->movi(TRegH(IDX(vmm_aux0)), 0);
+        h->fmaxnm(vmm_src.h, vmm_src.h, TRegH(IDX(vmm_aux0)));
```

Comment:
```suggestion
        h->movi(vmm_aux0.h, 0);
        h->fmaxnm(vmm_src.h, vmm_src.h, vmm_aux0.h);
```
Do you think we could simplify it to this, or would this break?

Reply:
I believe this only works on TReg (i.e. VReg or ZReg) objects, whereas all the `vmm_aux*` variables were declared as TRegS unfortunately. 

I could create a `vmm_aux_h` variable instead if you think that's clearer than the current version:
```suggestion
        const auto &vmm_aux_h = TRegH(IDX(vmm_aux1));
        h->movi(vmm_aux_h, 0);
        h->fmaxnm(vmm_src.h, vmm_src.h, vmm_aux_h);
```

Reply:
Thanks for the clarification. The refactoring can go as a separate PR.